db/migrate/20180711120130_add_missing_ems_id_to_switch.rb

```diff
@@ -0,0 +1,42 @@
+class AddMissingEmsIdToSwitch < ActiveRecord::Migration[5.0]
+  class ExtManagementSystem < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  class Host < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+
+    belongs_to :ext_management_system, :foreign_key => "ems_id"
+
+    has_many :host_switches
+    has_many :switches, :through => :host_switches
+  end
+
+  class HostSwitch < ActiveRecord::Base
+    belongs_to :host
+    belongs_to :switch
+  end
+
+  class Switch < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+
+    belongs_to :ext_management_system, :foreign_key => "ems_id"
+
+    has_many :hosts, :through => :host_switches
```

Comment:
This can get really nasty in migrations because Rails doesn't properly fully-qualify the class for this relationship.

Reply:
this is used only in spec, the migration is using raw SQL